rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -389,7 +372,7 @@ def _user_intent_from_step(
     ) -> Tuple[Text, Optional[Text]]:
         user_intent = step.get(KEY_USER_INTENT, "").strip()
 
-        if not user_intent:
+        if not user_intent and KEY_USER_MESSAGE not in step:
```

Comment:
Does the rest of this method gracefully handle there being no intent?

Reply:
We have tests for it and they pass 🤷‍♂️